src/web/transforms/images.ts

```diff
@@ -1,12 +1,98 @@
+import { oxaLinkToId, VersionId } from '@curvenote/blocks';
 import { GenericNode, selectAll } from 'mystjs';
+import fetch from 'node-fetch';
+import fs from 'fs';
 import path from 'path';
+import mime from 'mime-types';
+import { createHash } from 'crypto';
 
-import { Root } from './types';
+import { Root, TransformState } from './types';
+import { Options } from '../types';
 
-export function transformImages(mdast: Root) {
+export function serverPath(opts: Options) {
+  const buildPath = opts.buildPath || '_build';
+  return `${buildPath}/web`;
+}
+
+export function publicPath(opts: Options) {
+  return path.join(serverPath(opts), 'public');
+}
+
+export function imagePath(opts: Options) {
+  return path.join(publicPath(opts), '_static');
+}
+
+function hashUrl(url: string) {
+  return createHash('md5').update(url).digest('hex');
+}
+
+function isUrl(url: string) {
+  return url.toLowerCase().startsWith('http:') || url.toLowerCase().startsWith('https:');
+}
+
+export async function transformImages(mdast: Root, state: TransformState) {
   const images = selectAll('image', mdast) as GenericNode[];
-  images.forEach((image) => {
-    // assumes all images referenced in myst content are moved to public/_static
-    image.url = `/_static/${path.basename(image.url)}`;
-  });
+  await Promise.all(
```

Comment:
not awaiting here and returning this promise is also an option *if* you can await on `transformImages` ok... which it look like you are?